### PR TITLE
[JBPM-9790] Add integration test for emitter admin operations

### PR DESCRIPTION
**Pull request by gmunozfe**
**JIRA**: [JBPM-9790](https://issues.redhat.com/browse/JBPM-9790)

### CODE REVIEW

File: kie-spring-boot/kie-spring-boot-samples/jbpm-spring-boot-sample-basic/src/test/java/org/jbpm/springboot/samples/events/emitters/CountDownLatchEmitter.java [hunk 1]
```diff
@@ -41,13 +42,13 @@ public CountDownLatch getCountDownLatch() {
...
         log.info("deliver {}", data);
-        latch.countDown();
+        countDownIfTaskInstanceView(data);
     }
 
     @Override
     public void apply(Collection<InstanceView<?>> data) {
         log.info("apply {}", data);
-        latch.countDown();
+        countDownIfTaskInstanceView(data);
```
Comment (afalhambra) [inline kie-spring-boot/kie-spring-boot-samples/jbpm-spring-boot-sample-basic/src/test/java/org/jbpm/springboot/samples/events/emitters/CountDownLatchEmitter.java:51]: shouldn't we a similar logic or approach to the `drop` method to keep this class consistent? wdyt?

https://github.com/kiegroup/droolsjbpm-integration/blob/d7ef89dc56984e471c5c19237d5391a5818244eb/kie-spring-boot/kie-spring-boot-samples/jbpm-spring-boot-sample-basic/src/test/java/org/jbpm/springboot/samples/events/emitters/CountDownLatchEmitter.java#L57

**Review by afalhambra — APPROVED** [review]
Nice one @gmunozfe! only minor comments

**Review by fjtirado — APPROVED** [review]

File: kie-spring-boot/kie-spring-boot-samples/jbpm-spring-boot-sample-basic/src/test/java/org/jbpm/springboot/samples/EventEmitterTest.java [hunk 2]
```diff
@@ -1,12 +1,16 @@
 package org.jbpm.springboot.samples;
```
[the remark this answers is not in the document]
Comment (gmunozfe) [reply kie-spring-boot/kie-spring-boot-samples/jbpm-spring-boot-sample-basic/src/test/java/org/jbpm/springboot/samples/EventEmitterTest.java:17]: done

File: kie-spring-boot/kie-spring-boot-samples/jbpm-spring-boot-sample-basic/src/test/java/org/jbpm/springboot/samples/events/emitters/CountDownLatchEmitter.java [hunk 3]
```diff
@@ -7,6 +7,7 @@
 import org.jbpm.persistence.api.integration.EventEmitter;
```
[the remark this answers is not in the document]
Comment (gmunozfe) [reply kie-spring-boot/kie-spring-boot-samples/jbpm-spring-boot-sample-basic/src/test/java/org/jbpm/springboot/samples/events/emitters/CountDownLatchEmitter.java:23]: done

File: kie-spring-boot/kie-spring-boot-samples/jbpm-spring-boot-sample-basic/src/test/java/org/jbpm/springboot/samples/events/emitters/CountDownLatchEmitter.java [hunk 4]
```diff
@@ -68,4 +69,13 @@ public void close() {
         log.info("closing Event Emitter");
         latch.countDown();
     }
+    
+    protected void countDownIfTaskInstanceView(Collection<InstanceView<?>> data) {
+        if (data.stream().filter(instanceView -> instanceView instanceof TaskInstanceView).count() >= 1) {
```
[the remark this answers is not in the document]
Comment (gmunozfe) [reply kie-spring-boot/kie-spring-boot-samples/jbpm-spring-boot-sample-basic/src/test/java/org/jbpm/springboot/samples/events/emitters/CountDownLatchEmitter.java]: changed

File: kie-spring-boot/kie-spring-boot-samples/jbpm-spring-boot-sample-basic/src/test/java/org/jbpm/springboot/samples/events/emitters/CountDownLatchEmitter.java (hunk 1 above)
Comment (gmunozfe) [reply kie-spring-boot/kie-spring-boot-samples/jbpm-spring-boot-sample-basic/src/test/java/org/jbpm/springboot/samples/events/emitters/CountDownLatchEmitter.java:67]: makes sense, done

**Comment by gmunozfe** [comment]
jenkins retest this

**Comment by gmunozfe** [comment]
jenkins retest this

**Comment by gmunozfe** [comment]
jenkins retest this

**Comment by gmunozfe** [comment]
jenkins retest this

**Comment by gmunozfe** [comment]
jenkins retest this

**Comment by gmunozfe** [comment]
@mareknovotny could you merge?

**Pull request merged**